CryptoPrices/Sources/Data/Sources/DataMocks/MockCoinAPIProtocol.swift

```diff
@@ -0,0 +1,83 @@
+// Generated using Sourcery 1.9.2 — https://github.com/krzysztofzablocki/Sourcery
+// DO NOT EDIT
+// swiftlint:disable all
+
+import Foundation
+import NetworkCore
+
+public class MockCoinAPIProtocol: CoinAPIProtocol {
```

Comment:
I think we should add this file to `.gitignore` as I am wondering if we really need to register this file to Git or not. This is generated code anyway, right? What do you think? 🤔

Reply:
I was actually thinking about this while implementing, here are my thoughts 🙏:
- I think they are the _generated_ codes but they are an integrity part of the package, bundled into one once they're compiled. Of course, these are our codes (in the sense that written by us), packed in local packages, so we can update any time, build time or run time. But what if we treat them like third-party packages? When building we build from source (if open) or import binary one (if closed), I think most of the time (if not all) they don't generate any new files after we import them to our project.
- I'd like to think from another perspective 😄: Imagine we don't have any code generator tool (`Sourcery` or any other), we would add `Swift` files with those mock types written by hand and still add to version control as normal. Now we have `Sourcery` (thank God) we just have _not_ to do that by hand, but everything else is still the same. The codes still are part of the implementation. They are still the public, exposed types that other packages can import and use.
- For the _conflict_ part, that is the reason I generate code in a separate file per type, not in one big file. So for e.g. 2 developers at the same time add 2 new protocols, say `protocol UseCaseA` and `protocol UseCaseB`. In that case, there would be 2 separate files named `MockUseCaseA.swift` and `MockUseCaseB.swift` being generated, not being added in the same file (even in the same line). Even in the case that they both modify the same existing protocol, the chance of conflict is quite low and much easier to resolve.

That said, I just tried to `gitignore` the generated files in the local machine and the project still builds and runs as before. So either way is perfectly fine to me and I'm happy to update. But I'd to get more of your idea 🙏 What do you think?

Reply:
IMO, for a more significant project, putting these files into `.gitignore` would be a better practice. But for this project's scope, both approaches may show little difference in effect :D.

Reply:
@Thieurom to be honest, I think your thought directions are a way far off of my original intention on why we should add it to `.gitignore` here 🙏 Just to clarify a bit more, the topic is mainly on the `Git` storage usage question that should we push the generated files to the repositories or not? To me, the answer is quite clear that it is unnecessary to push generated code to the Git repo because it will be a waste of space and the generated code can be scaled up super quickly. 🙇‍♂️

For our project, at this scale, @nkhanh44 is right that it is not an immediate concern, but I think we should always strive for best practices at the initial phases so that we don't have to spend extra efforts to clean up later on, that would be my take 😁 Anyhow, I think the decision is still belong to the author to make 👍

Reply:
@minhnimble  @nkhanh44 Thank you guys, I really appreciate your feedback 🙏 

I decided to put those files in `.gitignore`, meaning that initially, those generated files are not parts of the module. They will be after the first build though. However, there was an error:

<img width="1487" alt="Screenshot 2022-12-26 at 09 12 12" src="https://user-images.githubusercontent.com/17972674/209490762-bf4dd809-52b1-4fb3-8321-69c59cee3168.png">

This happened because when compiling the `Home` package, it needs the `MockMyCoinsUseCaseProtocol` from the `Domain` package. However, the `Domain` package finishes the build process while the `MockMyCoinsUseCaseProtocol.swift` is generated or saved on the file system. It's kind of _not waiting_ for the files to be done generating. I realized that I had a mistake:
```swift
"--output",
context.package.directory.appending("Sources", "\(context.package.displayName)Mocks"),
```
I put the generated files in a _custom_ directory that is outside of the target that is currently being built (another target name "DomainMocks"). Those files wouldn't be the sources of the target, the compiler wouldn't compile those files. So the error occurred.

Actually, besides the `Sources` directory, each Swift package or target has another directory that the compiler will search while building, that is the `pluginWorkDirectory`. Here is a little more detail on [Package Manager Extensible Build Tools proposal](https://github.com/apple/swift-evolution/blob/main/proposals/0303-swiftpm-extensible-build-tools.md): 
>Every plugin invocation is passed the path of a directory in TargetBuildContext.pluginWorkDirectory. A plugin would usually create a separate subdirectory of this directory for each prebuild command it creates, and the command would be configured to write its output files into that directory. The plugin may also create other subdirectories for cache files and other file system content that either it or the command needs. If these additional files need to be available to the command, the plugin would construct the command line to include their paths, for the command to use.
Examples of plugins that need to use prebuild commands include SwiftGen and other tools that need to see all the input files, and whose set of output files is determined by the contents (not just the paths) of the input files. Such a plugin usually generates just one command and configures it with an output directory into which all generated sources will be written.

Here is an example of the `pluginWorkDirectory` path while building on my machine:

<img width="978" alt="Screenshot_2022-12-26_at_09_52_18" src="https://user-images.githubusercontent.com/17972674/209493760-41ecc563-e9cc-428d-a0d9-0ea41bb59827.png">

So I updated with the following:
```swift
let generatedDirectory = context.pluginWorkDirectory.appending("SourceryGenerated")
//
"--output",
generatedDirectory,
//
```
Actually, all the references in the PR's description follow this config/approach but I was too hurry to check them carefully 😓
So now it:
- Fix the build issue
- The generated files are not visible to the Xcode's navigator which is good cause we rarely bother with the content or what's going on in them. We can still jump to the definition if we want to. But we can't accidentally modify them (at least in Xcode) which is another good!
- Putting those in `.gitignore` is not necessary anymore since they belong to the sandbox directory.